src/NServiceBus.ContainerTests/When_using_nested_containers.cs

```diff
@@ -135,8 +113,41 @@ public void Instance_per_call_components_should_not_cause_memory_leaks()
                 var upperLimitBytes = 200*1024;
                 Assert.That(after - before, Is.LessThan(upperLimitBytes), "Apparently {0} consumed more than {1} KB of memory", builder, upperLimitBytes/1024);
             }
+        }
+
+        [Test]
+        [Explicit]
+        public void Instance_per_uow_components_should_not_cause_memory_leaks()
```

Comment:
That's exactly what I'm saying. This test spins up a child container and the class which is registered as a UOW in the parent container. So the child container should return a singleton. This test does this 20000 times, so there should be 20000 returning the same instance.

The test above this does the build from the parent container. This test builds from the child container.